backend/dal/Repositories/ActivityRepository.cs

```diff
@@ -41,13 +41,13 @@ public PimsActivityInstance GetById(long id)
             return this.Context.PimsActivityInstances
                 .Include(r => r.ActivityTemplate).ThenInclude(y => y.ActivityTemplateTypeCodeNavigation)
                 .AsNoTracking()
-                .FirstOrDefault(x => x.ActivityInstanceId == id);
+                .FirstOrDefault(x => x.ActivityInstanceId == id) ?? throw new KeyNotFoundException();
```

Comment:
Returning an empty object if not found sounds reasonable. The consumer could check for null or if the returned one has an id. As it is, it forces the consumer to handle a throw.

Reply:
I don't disagree, but this is not the pattern that we have been using. Also, I already had to fix a related issue in the research file where an empty response was being returned but the consumer forgot to check for it and just assumed the response was valid. I'd rather show this as an error, because a consumer should never request an id that doesn't exist, that indicates something went wrong. These endpoints aren't designed to be for searches.

Reply:
I see. Lets have a chat on friday to figure out which pattern we want to follow